### PR TITLE
Status API - Add build_id to Event::RepoPublished

### PR DESCRIPTION
**Pull request by dmarcoux**
Including changes from #5561 as discussed with @mlschroe

### CODE REVIEW

**Comment by dmarcoux** [comment]
[P1: Status API for Repository](https://trello.com/c/luTPbaqr/824-p1-status-api-for-repository)

File: src/api/app/models/update_notification_events.rb [hunk 1]
```diff
@@ -14,6 +14,10 @@ def create_events
         e.elements('data') do |d|
           data[d['key']] = d['_content']
         end
+        puts type
```
Comment (Ana06) [inline src/api/app/models/update_notification_events.rb]: :eyes:

Comment (dmarcoux) [reply src/api/app/models/update_notification_events.rb]: All this stuff is in a WIP commit. The description says that it's blocked for now...

**Comment by Ana06** [comment]
@dmarcoux 

> All this stuff is in a WIP commit. The description says that it's blocked for now...

it would be nice to know why it is a work in progress, why did you submit the PR? do you want to get feedback? are you trying it? :confused:

File: src/api/app/models/update_notification_events.rb (hunk 1 above)
Comment (dmarcoux) [reply src/api/app/models/update_notification_events.rb]: I didn't want to forget this when working on it this week. This is why I commited it.

**Comment by dmarcoux** [comment]
Feel free to review now!

**Comment by coolo** [comment]
This clearly deserves a test case - in one of the layers

**Comment by dmarcoux** [comment]
What would be the best way to test/validate this? I am not so sure.

**Comment by hennevogel** [comment]
> What would be the best way to test/validate this? I am not so sure.

Validate that a `Event::RepoPublished.payload` has those three keys?

**Comment by coolo** [comment]
But triggering this event would mean triggering it backend. I wonder why we track the payload_keys at all. It was useful to collect this information, but I wonder if we need to double check the backend at all by now.

File: src/api/spec/models/event/repo_published_spec.rb [hunk 2]
```diff
@@ -0,0 +1,9 @@
+require 'rails_helper'
+
+RSpec.describe Event::RepoPublished do
+  describe 'payload' do
+    subject { Event::RepoPublished.new({ project: 'project', repo: 'repo', buildid: 'buildid' }).payload }
+
+    it { is_expected.to eq({ 'project' => 'project', 'repo' => 'repo', 'buildid' => 'buildid' }) }
```
Comment (hennevogel) [inline src/api/spec/models/event/repo_published_spec.rb]: What are you testing? ruby variable assignment?

**Comment by hennevogel** [comment]
Alright validating this is discussed in #5619 and we reverted to the old way in 93953d869013369c5deac7c2fe1659da79080dd8

**Pull request merged**